### PR TITLE
Added consensus moa analysis files

### PR DESCRIPTION
**Pull request by AdeboyeML**
@gway I have added the jupyter notebook that contains the code for the analysis on the consensus datasets to get the moa sizes/values per doses(1-7), I have also included the results of the analysis i.e. the heatmap plots and the csv files (folders).

### CODE REVIEW

**Comment by gwaygenomics** [comment]
this looks great @AdeboyeML - a couple of immediate things to change, before I take a deeper dive.

1. Let's move this entire analysis into a folder called `1.data-exploration`.
2. Let's be sure to keep data extensions on csv files. For example, `moa_sizes_consensus_datasets/consensus_median_moa_sizes` should be `moa_sizes_consensus_datasets/consensus_median_moa_sizes.csv`.

The reason for 1 is that we want to perform several analyses in this repo, and this is just one component. Each new analysis will belong in an analysis submodule.

**Comment by AdeboyeML** [comment]
@gwaygenomics Thanks for the observations. 

I have created the `1.Data-exploration`  folder (putting all files and sub-folders into the folder), and also updated the notebook with the null distribution analysis and I have also saved all the dataframe outputs as `.csv files`.

The `.csv files` has both the  median_scores and p_values dataframes for each consensus dataset per dose (for each of the 231 MOAs with more than one compound).

**Comment by gwaygenomics** [comment]
Looks great @AdeboyeML - sorry this has taken me so long to review. I am just about ready to approve.

## Two major things:

1. It looks like there is a large number of cells in the notebook that are not executed. Can you either execute the cell or delete the cells? Jupyter notebooks need to be executed sequentially, and if I am viewing a notebook without sequential execution, I have extreme doubts about the data analysis. (not saying that I doubt this one, just in general it is suboptimal practice)
2. Can you add in the figures you generated in lincs-profiling-comparison/issues/2? They look great, and are not currently captured in this PR. I would also switch to histograms from density plots (in the seaborn jointplot) since the density tails extend far beyond p < 0 (which doesn't make sense).

## Other pull request advice

* I definitely appreciate the scope of this PR. It is not huge. I have reviewed big pull requests in the past and they are not fun! However, the jupyter notebook is quite long. In the future, we should consider including fewer analyses in a single jupyter notebook, and instead number the notebooks if they require sequential execution. (e.g. `consensus_moa_sizes_per_dose_analysis.ipynb` becomes `0.consensus_moa_sizes_per_dose_analysis.ipynb` and one other script to visualize results, or something)
* It is also good practice to include the nbconverted file from the `.ipynb` files. These nbconverted files can be generated with the following command: `jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/nbconverted *.ipynb`. The nbconverted files **are so much easier to review** than .ipynb files.

## Advice for myself

* I think I might be needlessly holding you up on performing the next analyses, and I don't want to be! We value good and reproducible code extremely highly, but I need to be aware of when this search for perfection slows things up too much!

**Comment by shntnu** [comment]
Dropping in with a minor comment: Please add the column name for the first column (`moa`) to the CSV files
e.g. https://raw.githubusercontent.com/broadinstitute/lincs-profiling-comparison/9bc5db8167674e2c8bec5cee3fcc043117acfbf6/1.Data-exploration/moa_sizes_consensus_datasets/median_dmso_null_p_values.csv

**Comment by gwaygenomics** [comment]
@AdeboyeML - we should prioritize merging in this contribution early next week. Here is a todo list:

- [ ] Remove unexecuted cells and non-essential comments
- [ ] Make sure all cells are executed in order
- [ ] Make sure all columns have names in all data that is output (nice catch @shntnu)

I see people forgetting to add column names for indices all the time, and it bugs me! It is also a very easy fix:

```python
data_moa_values  = data_moa_values.reset_index().rename({"index": "moa"}, axis="columns")
save_to_csv(data_moa_values, 'moa_sizes_consensus_datasets', 'modz_moa_median_scores.csv')
```

**Comment by AdeboyeML** [comment]
@gwaygenomics - thanks for the observations and major points highlighted.


-  I have done all necessary corrections to the jupyter notebooks and csv files. I have re-pushed new notebooks and files to the lincs-profiling-comparison repo containing notebooks for different profiling technologies and have also created new independent notebooks for different use cases.

File: 1.Data-exploration/L1000/L1000_moa_median_scores_calculation.ipynb [hunk 1]
```diff
@@ -0,0 +1,1477 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Consensus Signatures\n",
+    "\n",
+    "A consensus signature can be defined as a perturbation-specific summary profile acquired by aggregating replicate level information.\n",
+    "\n",
+    "\n",
+    "#### Level 5 - Replicate-consensus signatures (MODZ)\n",
+    "L1000 experiments are typically done in 3 or more biological replicates. We derive a consensus replicate signature by applying the\n",
```
Comment (gwaygenomics) [inline 1.Data-exploration/L1000/L1000_moa_median_scores_calculation.ipynb:13]: I knew that I recognized this text! 

When something like this is copy and pasted, we _must_ add a citation to the source. In this case add to the markdown:

From [Subramanian et al 2017](https://doi.org/10.1016/j.cell.2017.10.049):

> L1000 experiments are typically done in 3 or more biological replicates. We derive a consensus replicate signature by applying the moderated z-score (MODZ) procedure as follows. First, a pairwise Spearman correlation matrix is computed between the replicate signatures in the space of landmark genes with trivial self-correlations being ignored (set to 0). Then, weights for each replicate are computed as the sum of its correlations to the other replicates, normalized such that all weights sum to 1. Finally, the consensus signature is given by the linear combination of the replicate signatures with the coefficients set to the weights. This procedure serves to mitigate the effects of uncorrelated or outlier replicates, and can be thought of as a ‘de-noised’ representation of the given experiment’s transcriptional consequences.

Make sure to include a URL to the DOI in the paper reference

File: 1.Data-exploration/L1000_cell_painting_comparison/.ipynb_checkpoints/L1000_CP_comparison_visualization-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,221 @@
+{
```
Comment (gwaygenomics) [inline 1.Data-exploration/L1000_cell_painting_comparison/.ipynb_checkpoints/L1000_CP_comparison_visualization-checkpoint.ipynb:1]: `.ipynb_checkpoints` files should not be included. Please remove this file and add `.ipynb_checkpoints/` to your `.gitignore`.

**Review by gwaygenomics — APPROVED** [review]
A couple critical comments. Please address before merging.

Also, please make sure to run:

```bash
jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/nbconverted *.ipynb
```

In your terminal, in directories where there are `.ipynb` files. The command will create a new `.py` file corresponding to the `.ipynb` notebook. Making in line comments in the `.ipynb` JSON is very tough.

**Comment by gwaygenomics** [comment]
once you address these three things: 

- [ ] Add citation
- [ ] Remove `.ipynb_checkpoints` file and add to `.gitignore` file (example [here](https://github.com/broadinstitute/cell-health/blob/master/.gitignore#L30))
- [ ] Add `.py` files for corresponding `.ipynb` files

Go ahead and merge.

Nice work! 🎉

**Comment by AdeboyeML** [comment]
@gwaygenomics all highlighted points have been addressed and corrected.

**Comment by gwaygenomics** [comment]
LGTM 👍 please merge at will

**Comment by AdeboyeML** [comment]
@gwaygenomics I do not think I have the permission to merge the pull request. 

> Only those with write access to this repository can merge pull requests.

**Comment by gwaygenomics** [comment]
ah, right. Just made you admin. You need to accept in an email, then you should be able to merge. Thanks for pointing this out!

**Pull request merged**